src/main/java/com/redhat/devtools/lsp4ij/internal/CompletableFutures.java

```diff
@@ -132,6 +150,10 @@ public static void waitUntilDone(@Nullable CompletableFuture<?> future,
                     // This wait can block IJ, here we stop the wait (and we could lose some LSP feature)
                     throw new CancellationException("Some projects are indexing");
                 }
+                if (pendingFutures != null) {
+                    pendingFutures.add(future);

```

Comment:
I wonder if it should be better to store pendingFutures in the editor data and use an HashSet to avoid adding the same future?